src/utils/specific_label_utils.py

```diff
@@ -119,6 +95,129 @@ def is_insertion_or_replication(word, correct_words_list):
+
+    typo_word_map = {}
+
+    for word in unique_flawed_words:
+        if label_func:
+            typo_word_map[word] = label_func(word, min_value, max_value) 
+        else:
+            typo_word_map[word] = label_number_with_ocr_or_typo(word, min_value, max_value)
+
+    # Remap results back to the original indices
+    for index, word in flawed_words_series.items():
+        if typo_word_map.get(word, False):
+            label_column.loc[index] = typo_word_map[word]
+
+    return label_column
+
+def label_number_with_ocr_or_typo(word: str | int | float, min_value: float = None, max_value: float = None):
+    """
+    Method, which can be used to label numbers with errors, in case there is no column-specific labeling function for numbers defined.
+    It distinguishes between typos and OCRs for numbers.
+    """
+    
+    word = str(word)
+    if word == 77:
+        print("77 from label_number_with_error")
+    if word[-1] in [".", ",", "-"]: # a number like 8743. that ends with a dot is likely an OCR
+        return ErrorType.OCR.value
+
+    if word[0] == "0" and word[1] != ".": # a number like 08.1 that starts with a 0 is likely an OCR
+        return get_label_for_number_with_0_prefix(word, min_value, max_value)
+
+    if " " in word:
+        return ErrorType.OCR.value
+
+    if is_likely_ocr(word):
+        return ErrorType.OCR.value
+    
+    if contains_letter(word): # this covers all letters that are not OCRs (because we already checked for OCR letters), therefore we can label them as typos
+        return ErrorType.TYPO.value
+
+    if min_value is not None and max_value is not None:
+        if is_replaced_ocr_in_range(word, min_value, max_value): # values like 77 would be replaced to 17, 71 or 11, which could be in range and therefore labeled as OCR instead of typo
+            return ErrorType.OCR.value
+
+    # TODO: When improveing the generic labeling of a number column, print, which values are labeled as Typos to see if they are really typos or if we can label them as OCRs.
+    # print("Typo: ", word)
+
+    # TODO: Check if it is necessary to differentiate further. We would need concrete examples of where a typo was introduced in a number column.
+    return ErrorType.TYPO.value
+
+
+def contains_letter(word: str) -> bool:
+    """
+    Check if the given word contains any letter (a-z or A-Z).
+    """
+    return any(char.isalpha() for char in word)
+
+
+def get_label_for_number_with_0_prefix(word, min_value: float = None, max_value: float = None):
+    try:
+        num = float(word[1] + word[0] + word[2:]) # switch first two characters
+        if (min_value is not None and min_value <= num) and (max_value is not None and num <= max_value):
+            return ErrorType.TYPO.value
+    except ValueError:
+        return ErrorType.OCR.value
+    return ErrorType.OCR.value
+
+def is_likely_ocr(word: str | int | float):
+    aggregated_ocr_letters = []
+    for key in OCR_DICT.keys():
+        if key.isalpha():
```

Comment:
Add a comment justifying why this check is neccessary, took me longer than I would admit to understand haha